### DISCUSSION TITLE
Kong 2.8.1 memory rise with increase in number of upstream

### DISCUSSION THREAD

**Discussion by sunny-deo-incontact** [Kong/kong#10330] [Help]
Hi Folks,

We have observed following behavior in Kong 2.8.1.

_**1st run**_
No of upstream server: 60
Requests per second: 200
Instance size : T2.Medium
Deployment done through docker image.

We dynamically set upstream using kong.service.set_target() method.

We observe that memory rises continuously, and instance runs out of memory.

**_2nd run_**
No of upstream server: 5
Requests per second: 200
Instance size : T2.Medium
Deployment done through docker image.

We dynamically set upstream using kong.service.set_target() method.

We observe that memory rises up to certain limit but does not run out of memory.

**Note**: We are not using upstream object.

Can anyone please help us understand why this is happening?

**Comment by ghost** [comment]
the different between 1st run and 2nd run just only `No of upstream server`?

> We dynamically set upstream using kong.service.set_target() method.

Can you add details?  How to add 60 upstream server?

**Reply by sunny-deo-incontact** [reply]
We have a custom plugin which fetches Upstream server key from Bearer token (bearer token contains unique server key in encoded format). 
We have a Json file where we maintain Upstream server key and its value (in this case upstream server domain name and port for ex. www.google.com:443). 
We have configured 60 different upstream servers domain names based on 60 different keys in JSON format. 
Based on this key's value we set the target using kong.service.set_target() for each request that we receive. 

When we run with 6 upstream servers in file it runs without any issue, however if we keep increasing upstream servers, we start seeing rise in memory usage.

**Reply by ghost** [reply]
Then you need to provide minimal reproducible custom code.

I guess it's possible that set_target allocated some resources but didn't free them.

Anyway, if you provide a detailed reproducible demo, I'm willing to look into it.
